src/lib.rs

```diff
@@ -0,0 +1,886 @@
+// licenses.
+
+#![crate_name = "ldk_lite"]
+
+//! A library providing a simplified API for the Lightning Dev Kit. While LDK itself provides a
+//! highly configurable and adaptable interface, this API champions simplicity and ease of use over
+//! configurability. To this end, it provides an opionated set of design choices and ready-to-go
+//! default modules, while still enabling some configurability when dearly needed by the user:
+//! - Chain data is accessed through an Esplora client.
+//! - Wallet and channel states are persisted to disk.
+//! - Gossip is retrieved over the P2P network.
+
+#![deny(missing_docs)]
+#![deny(broken_intra_doc_links)]
+#![deny(private_intra_doc_links)]
+#![allow(bare_trait_objects)]
+#![allow(ellipsis_inclusive_range_patterns)]
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+
+mod access;
+mod error;
+mod event;
+mod logger;
+mod util;
+
+use access::LdkLiteChainAccess;
+pub use error::LdkLiteError as Error;
+pub use event::LdkLiteEvent;
+use event::LdkLiteEventHandler;
+
+#[allow(unused_imports)]
+use logger::{
+	log_error, log_given_level, log_info, log_internal, log_trace, log_warn, FilesystemLogger,
+	Logger,
+};
+
+use lightning::chain::keysinterface::{InMemorySigner, KeysInterface, KeysManager, Recipient};
+use lightning::chain::{chainmonitor, Access, BestBlock, Confirm, Filter, Watch};
+use lightning::ln::channelmanager;
+use lightning::ln::channelmanager::{
+	ChainParameters, ChannelManagerReadArgs, SimpleArcChannelManager,
+};
+use lightning::ln::peer_handler::{IgnoringMessageHandler, MessageHandler, SimpleArcPeerManager};
+use lightning::ln::{PaymentHash, PaymentPreimage, PaymentSecret};
+use lightning::routing::gossip;
+use lightning::routing::gossip::P2PGossipSync;
+use lightning::routing::scoring::ProbabilisticScorer;
+
+use lightning::util::config::UserConfig;
+use lightning::util::ser::ReadableArgs;
+
+use lightning_background_processor::BackgroundProcessor;
+use lightning_background_processor::GossipSync as BPGossipSync;
+use lightning_persister::FilesystemPersister;
+
+use lightning_net_tokio::SocketDescriptor;
+
+use lightning_invoice::utils::DefaultRouter;
+use lightning_invoice::{payment, Currency, Invoice};
+
+use bdk::blockchain::esplora::EsploraBlockchain;
+use bdk::blockchain::{GetBlockHash, GetHeight};
+use bdk::database::MemoryDatabase;
+
+use bitcoin::hashes::sha256::Hash as Sha256;
+use bitcoin::hashes::Hash;
+use bitcoin::secp256k1::PublicKey;
+use bitcoin::BlockHash;
+
+use rand::RngCore;
+
+use std::collections::HashMap;
+use std::fs;
+use std::net::SocketAddr;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::mpsc;
+use std::sync::{Arc, Mutex, RwLock};
+use std::thread;
+use std::time::{Duration, SystemTime};
+
+// TODO: Is MemoryDatabase okay to use?
+
+const CHANNEL_BUF_SIZE: usize = 1000;
+
+#[derive(Debug, Clone)]
+/// Represents the configuration of an `LdkLite` instance.
+pub struct LdkLiteConfig {
```

Comment:
Mh, not sure about this one, because this is really meant as a library, not a standalone application. So while we indeed currently assume a data directory, I think that we still want to keep the API methods as the main interface for setting/accessing the config options. That said, we should probably provide methods to read/persist `LdkLiteConfig` from/to disk to allow the user to recover the set values and init `LdkLite` through `LdkLiteBuilder::from_config`. I added an issue for this: https://github.com/lightningdevkit/ldk-lite/issues/4